wkcuber/wkcuber/export_wkw_as_nifti.py

```diff
@@ -0,0 +1,206 @@
+
+def export_layer_to_nifti(
+    wkw_file_path: Path,
+    bbox: Dict,
+    mag: Mag,
+    layer_name: str,
+    destination_path: Path,
+    name: str,
+    original_bbox_size: Optional[List[int]] = None,
+    offset_into_orginal_bbox: Optional[List[int]] = None,
+    bounding_box_crop: Optional[List[int]] = None,
+) -> None:
+    wk_ds = Dataset(wkw_file_path)
+    layer = wk_ds.get_layer(layer_name)
+    mag_layer = layer.get_mag(mag)
+
+    is_segmentation_layer = layer.category == LayerCategories.SEGMENTATION_TYPE
+
+    data = mag_layer.read(bbox["topleft"], bbox["size"])
+    data = data.transpose(1, 2, 3, 0)
+    logging.info(f"Shape with layer {data.shape}")
+
+    data = np.array(data)
+    if is_segmentation_layer and data.max() > 0:
+        factor = np.iinfo("uint8").max / data.max()
+        data = data * factor
+        data = data.astype(np.dtype("uint8"))
+
+    if original_bbox_size and offset_into_orginal_bbox:
+        padding_per_axis = []
+
+        for i in range(3):
+            if original_bbox_size[i] == data.shape[i]:
+                padding_per_axis.append((0, 0))
+            else:
+                left_pad = offset_into_orginal_bbox[i]
+                right_pad = (
+                    original_bbox_size[i] - offset_into_orginal_bbox[i] - data.shape[i]
+                )
+                padding_per_axis.append((left_pad, right_pad))
+
+        padding_per_axis.append((0, 0))
+        print(padding_per_axis)
+        data = np.pad(data, padding_per_axis, mode="constant", constant_values=0)
+
+        if bounding_box_crop is not None:
+            assert (
+                len(bounding_box_crop) == 6
+            ), "bounding box crop needs to have 6 parameters"
+
+            print(f"Using Bounding Box {bounding_box_crop}")
+
+            data = data[
+                bounding_box_crop[0] : bounding_box_crop[0] + bounding_box_crop[3],
+                bounding_box_crop[1] : bounding_box_crop[1] + bounding_box_crop[4],
+                bounding_box_crop[2] : bounding_box_crop[2] + bounding_box_crop[5],
+            ]
+
+    img = nib.Nifti1Image(data, np.eye(4))
+
+    destination_file = str(destination_path.joinpath(name + ".nii"))
+
+    logging.info(f"Writing to {destination_file} with shape {data.shape}")
+
+    nib.save(img, destination_file)
+
+
+def export_nifti(
+    wkw_file_path: Path,
+    bbox: Optional[BoundingBox],
+    mag: Mag,
+    destination_path: Path,
+    name: str,
+    original_bbox_size: Optional[List[int]] = None,
+    offset_into_orginal_bbox: Optional[List[int]] = None,
+    bounding_box_crop: Optional[List[int]] = None,
+) -> None:
+    wk_ds = Dataset(wkw_file_path)
+
+    for layer_name, layer in wk_ds.layers.items():
+        if bbox is None:
+            bbox_dict = {
+                "topleft": layer.bounding_box.topleft,
+                "size": layer.bounding_box.size,
+            }
+        else:
+            bbox_dict = {"topleft": list(bbox.topleft), "size": list(bbox.size)}
```

Comment:
why do you convert the BoundingBox instance to a dict? Passing around the object seems better to me. It also provides all the necessary information plus helpful methods, such as `contains_bbox`.

Reply:
It probably made sense a long time ago. Changed it now.